sphinx_needs/roles/need_ref.py

```diff
@@ -2,16 +2,53 @@
 from sphinx.application import Sphinx
 from sphinx.util.nodes import make_refnode
 
+from sphinx_needs.nodes import Need
 from sphinx_needs.errors import NoUri
 from sphinx_needs.logging import get_logger
 from sphinx_needs.utils import check_and_calc_base_url_rel_path, unwrap
 
+from collections.abc import Iterable
+
+from typing import Dict
+
 log = get_logger(__name__)
 
 
 class NeedRef(nodes.Inline, nodes.Element):
     pass
 
+def transform_need_to_dict(need: Need) -> Dict[str, str]:
```

Comment:
Do we maybe want to shift this function to a more generic place? I belive there is a reuse in other algorithms, too

Reply:
A need is a dictionary, but the value is sometimes a dictionary, list, ... . It is not nice to see this python syntax in teh output, so I translate a need in a dictionary of str, str. Why do I need a dictionary of str, str. This is the input to a str.format() function.

Reply:
In which cases do we have a `dict` as values of a need option?
And even if this is the case, are these values needed in a `need` representation?

We use jinja for all string-manipulations in Sphinx-Needs. And it would give the user more power, e.g. cutting long strings.
So I would like to stay with one, tested solution in the code and try to avoid a new one.

Reply:
I'm not sure if we talk about the same. In the current implementation of needref is this [line](https://github.com/useblocks/sphinxcontrib-needs/blob/master/sphinx_needs/roles/need_ref.py#L64).
`                link_text = app.config.needs_role_need_template.format(
                    title=title,
                    id=ref_id_complete,
                    type=target_need["type"],
                    type_name=target_need["type_name"],
                    status=target_need["status"],
                    content=target_need["content"],
                    tags=";".join(target_need["tags"]),
                    links=";".join(target_need["links"]),
                    links_back=";".join(target_need["links_back"]),
                )`
What I did:
Instead of doing the mapping everytime explicit (with maintance effort on all places, if a new option is added), I create the function `def transform_need_to_dict(need: Need) -> Dict[str, str]:` which is doing the mapping and use the current format function to create the string. See:
link_text = app.config.needs_role_need_template.**format**(

I'm not a jinja expert and so I cannot do the needed refactoring of the current needref to reflect all string manipulation with Jinja code.

Reply:
Argh I didn't know that this part is so "old" and is not using jinja implementation.
And before we introduce here a new way to use need-data inside strings, I would like to go directly with jinja.

There are several locations in the code, where it gets used. But in general this is how it works:
```python
from jinja2 import Template
ref_template = Template(ref_string, autoescape=True)
final_ref_name = content_template.render(**need_info)
 ```
So I think this should even reduce the lines of code and make really all the need-data available.
What do you think?